<!--
	Launch file for Baxter's hand-eye calibration
	
	After launching this file...
		
	SoCS, UoG. Gerardo Aragon-Camarasa & Aamir Khan. 09/2016
-->

<launch>
	
	<!-- Configuration file for camera calibration -->
	<param name="/glasgow_calibration/target" value="opencv" />
    <param name="/glasgow_calibration/marker_width" value="0.023" />
    <param name="/glasgow_calibration/marker_size_x" value="9" />
    <param name="/glasgow_calibration/marker_size_y" value="6" />
    <param name="/glasgow_calibration/max_error" value="2.0" />
    
    <!-- Where to save stuff -->
	<param name="/glasgow_calibration/outputImageDir" value="$(find calibration_glasgow)/calibrations/Images/" />
    <param name="/glasgow_calibration/outputCalibDir" value="$(find calibration_glasgow)/calibrations/" />
    
    <!-- Saving? 0: Only calib stuff is saved, 1: save processed images in hard disk plus calib stuff, 2: Save everything-->
    <param name="/glasgow_calibration/save_mode" value="2" type="int"/>
    <!-- Scale ratio in order to reduce the image while finding the inner corners-->
    <param name="/glasgow_calibration/resize_imgs_factor" value="1.0"/>

    <!-- Where to save the gripper to camera calibration file -->
    <param name="/glasgow_calibration/gHc_calibration_file" value="$(find calibration_glasgow)/" />
    
	<!-- Calibration service node -->
	<node pkg="calibration_glasgow" type="calibration_services" name="calibration_services" output="screen">
	</node>


	<!-- Messages node -->
	<!-- <include file="$(find kinect2_bridge)/launch/kinect2_bridge.launch">
    </include> -->
	<node pkg="twodto3d" type="twodto3d" name="twodto3d" output="screen">
	</node>
	<node pkg="twodto3d" type="transform_pub" name="transform_pub" output="screen">
	</node>
	
	<!-- For debug -->
	<param name="/glasgow_calibration/debug" type="bool" value="False" />
	
</launch>

